modules/core/src/main/scala/org/tessellation/infrastructure/rewards/Rewards.scala

```diff
@@ -74,10 +73,19 @@ object Rewards {
 
           val mintedAmount = getAmountByEpoch(epochProgress, config.rewardsPerEpoch)
 
-          allRewardsState
+          val rewards: F[SortedSet[RewardTransaction]] = allRewardsState
             .run(mintedAmount)
             .flatTap(validateState(mintedAmount))
             .map(toTransactions)
+
+          val oneTimeRewards = config.oneTimeRewards
+            .filter(_.epoch === epochProgress)
+            .map(otr => RewardTransaction(otr.address, otr.txnAmount))
+
+          // TODO should we handle case where Address already in rewards map and sum TransactionAmounts?
+          rewards.map(_.concat(oneTimeRewards))
+
+        // TODO Add test if this was correct place to add one-time rewards
```

Comment:
I'm not sure how it will affect unit tests. I believe we already check the total sum of rewards if it matches the expected value

Reply:
AFAIK he has retested the existing unit tests and added more, believe this is resolved?

Reply:
@marcinwadon I don't understand what you're asking me to test. When the one-time rewards is not-empty then those rewards show up for the epoch they're set for. When it's not that epoch it doesn't show up. I had unit tests in place showing it worked when it was non-empty.

Reply:
@marcinwadon Does adding this to my test in `RewardsSuite` address your concern? The test  fails if the one-time reward is during the same epoch but passes when it is not, as it's written here.
```
      for {
        oneTimeRewards <- // one-time reward in another epoch should not show up in results
          List(OneTimeReward(epoch.next, stardustNewPrimary, TransactionAmount(12345L))).pure[F]

        rewards = makeRewards(adjustedConfig.copy(oneTimeRewards = oneTimeRewards))
```